### ISSUE TITLE
plot is intialized abnormally on IE 11

### ISSUE THREAD

**Issue by jizg** [bokeh/bokeh#6217]
bokeh 0.12.6dev2, python 3.6.0, IE 11 on Windows 10

use weather example program under bokeh (example/app/weather) to do test on IE 11, the plot is initialized abnormally as follows. But after changing city by drop down list, the plot becomes correct. 
<img width="1136" alt="image" src="https://cloud.githubusercontent.com/assets/8054244/25556844/2299259a-2d37-11e7-94bd-d7240efbe585.png">

Python shows the following warning:
```
/Users/Jizg/anaconda/lib/python3.6/site-packages/bokeh/models/sources.py:91: BokehUserWarning: ColumnDataSource's columns must be of the same length. Current lengths: ('actual_max_temp', 366), ('actual_min_temp', 366), ('average_max_temp', 366), ('average_min_temp', 366), ('date', 732), ('left', 732), ('record_max_temp', 366), ('record_min_temp', 366), ('right', 732)
  "Current lengths: %s" % ", ".join(sorted(str((k, len(v))) for k, v in data.items())), BokehUserWarning))
/Users/Jizg/anaconda/lib/python3.6/site-packages/bokeh/models/sources.py:91: BokehUserWarning: ColumnDataSource's columns must be of the same length. Current lengths: ('actual_max_temp', 366), ('actual_min_temp', 366), ('average_max_temp', 366), ('average_min_temp', 366), ('date', 732), ('left', 732), ('record_max_temp', 366), ('record_min_temp', 366), ('right', 732)
  "Current lengths: %s" % ", ".join(sorted(str((k, len(v))) for k, v in data.items())), BokehUserWarning))
```
And the following error messages are displayed continuously even after the plot is displayed:
```
2017-04-29 23:58:17,329 received invalid integer in pong b''
Traceback (most recent call last):
  File "/Users/Jizg/anacnda/lib/python3.6/site-packages/bokeh/server/views/ws.py", line 167, in on_pong
    self.latest_pong = int(codecs.decode(data, 'utf-8'))
ValueError: invalid literal for int() with base 10: ''
````
And the browser console shows the following warning, same as python warning messages:
```
[bokeh] data source has columns of inconsistent lengths
```

It seems that length of some columns in datasource become 2 times more than others( 732 vs 366 in the previous example). But they should be the same. I cannot fix the reason.
ps. Everything is OK in Firefox and Chrome.

**Comment by bryevdv** [comment]
The `pong` messages on Window are harmless (though annoying) but we will look into the other issue as soon as we are able.

**Comment by mattpap** [comment]
Interestingly enough, the plot is initially rendered correctly and on second render during initialization gets messed up. This second render is completely unnecessary, but it should be only a performance penalty and the plot show render properly.

**Comment by seanlaw** [comment]
I am experiencing the same thing where I am running a bokeh server (on a Mac Mini) and hitting the server from IE on a Windows machine. The circle plot loads twice with the first being rendered correctly and then the browser quickly loads a second time with the circles completely rearranged. It complains that the X and Y coordinates from the CDS are twice as long as expected. All of the other CDS columns appear to be the correct length though. I get the same log as the OP.

**Comment by Minizinger** [comment]
Do you have any plans regarding this? I've encountered same issue with current project which doesn't allow for upgrade from version 0.12.4 (this doesn't happen there). I know that IE isn't officially supported anymore, however it's still actively used.

**Comment by gcmbr** [comment]
I have had identical issue with IE and bokeh server.  Error is about ColumnDataSource columns not of same length. Works fine in all other browsers.  Some columns are mistakenly read in as twice the true column data.

**Comment by cloud-rocket** [comment]
Any news on this one?

**Comment by cloud-rocket** [comment]
After some digging I found a workaround for this problem, which works with bokeh 0.12.13:

Instead of creating ColumnDataSource from pandas DataFrame directly I am casting it to dictionary before. (I think it prevents Bokeh from using Binary protocol falling back to JSON):

The mentioned weather example works if you replace line 29:
```
    return ColumnDataSource(data=df)
```
with
```
    return ColumnDataSource(data=df.reset_index().fillna('NaN').to_dict(orient="list"))
```

**Comment by bryevdv** [comment]
This was fixed at some point, win10, ie11, 1.1.0dev9 initial load:

<img width="1106" alt="Screen Shot 2019-03-08 at 9 58 59 PM" src="https://user-images.githubusercontent.com/1078448/54066929-7eaf8f00-41ed-11e9-825b-3b6d0fbb2e33.png">
